### ISSUE TITLE
"Cannot import 'libtorrent' due to 'invalid or missing encoding declaration" false positive

### ISSUE THREAD

**Issue by matkoniecz** [PyCQA/pylint#10849]
### Bug description
I get

E0001: Cannot import 'libtorrent' due to 'invalid or missing encoding declaration for '/home/mateusz/python_import_location/lib/python3.12/site-packages/libtorrent/__init__.cpython-312-x86_64-linux-gnu.so'' (syntax-error)

despite import working perfectly fine

### Command used

`pylint 1_obtain_osm_data.py` on file with

```
import libtorrent
```

content

Real project is at https://codeberg.org/matkoniecz/list_how_openstreetmap_can_be_improved_with_alltheplaces_data/src/branch/master/1_obtain_osm_data.py

Real pylint command was `pylint *.py --include-naming-hint=y --variable-rgx="^[a-z][a-z0-9]*((_[a-z0-9]+)*)?$" --argument-rgx="^[a-z][a-z0-9]*((_[a-z0-9]+)*)?$" --disable=R0917,E0401,W0201,R0902,C0103,C0301,C0114,C0115,C0116,C0121,W0613,R0911,R0912,R0913,R0915,C0302,C1803,R1710,W0719,R1713,R1705,C0411,W1514,E1136,W0101,W0611,R0904,E1124,R0801,W0706,C0413,R0903,R0402,C0325,C0206,R1724,R1737,R1714,R1702,W0621,C0209,R0914,R1730,W0105,W0603,W0108,W0511,C0201 --check-str-concat-over-line-jumps=y` 

and now is 

`pylint *.py --include-naming-hint=y --variable-rgx="^[a-z][a-z0-9]*((_[a-z0-9]+)*)?$" --argument-rgx="^[a-z][a-z0-9]*((_[a-z0-9]+)*)?$" --disable=R0917,E0401,W0201,R0902,C0103,C0301,C0114,C0115,C0116,C0121,W0613,R0911,R0912,R0913,R0915,C0302,C1803,R1710,W0719,R1713,R1705,C0411,W1514,E1136,W0101,W0611,R0904,E1124,R0801,W0706,C0413,R0903,R0402,C0325,C0206,R1724,R1737,R1714,R1702,W0621,C0209,R0914,R1730,W0105,W0603,W0108,W0511,C0201 --check-str-concat-over-line-jumps=y | grep -v libtorrent | grep -v 1_obtain_osm_data` as a workaround

### Pylint output

```python
************* Module 1_obtain_osm_data
1_obtain_osm_data.py:1:0: C0114: Missing module docstring (missing-module-docstring)
1_obtain_osm_data.py:1:0: C0103: Module name "1_obtain_osm_data" doesn't conform to snake_case naming style (invalid-name)
1_obtain_osm_data.py:1:0: E0001: Cannot import 'libtorrent' due to 'invalid or missing encoding declaration for '/home/mateusz/Documents/install_moje/shared_python_virtual_environment/lib/python3.12/site-packages/libtorrent/__init__.cpython-312-x86_64-linux-gnu.so'' (syntax-error)
1_obtain_osm_data.py:1:0: W0611: Unused import libtorrent (unused-import)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

it does not complain, like it does not complain about my other imports

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.2
Python 3.12.3 (main, Jan 22 2026, 20:57:42) [GCC 13.3.0]
```

### OS / Environment

QTerminal 1.4.0

`lsb_release -a` shows
```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.3 LTS
Release:        24.04
Codename:       noble
```

### Additional dependencies

```python
`libtorrent==2.0.11`
```

**Comment by Pierre-Sassoulas** [comment]
Syntax error are coming directly from the python interpreter, are you sure you're using python 3 and not python 2 in the failing command ?

**Comment by matkoniecz** [comment]
As far as I see `pylint` command is not failing, as far as I see command is successfully (and wrongly) claiming that I have syntax errors

I am pretty sure that I have no Python2 being involved, how can I recheck it?

Note that for other imports it has no such complaints

**Comment by Pierre-Sassoulas** [comment]
Use ``python3 -m pylint`` instead of ``pylint`` in the command you've shown in the issue description.

**Comment by matkoniecz** [comment]
also gives `E0001: Cannot import 'libtorrent' due to 'invalid or missing encoding declaration for '/home/mateusz/Documents/install_moje/shared_python_virtual_environment/lib/python3.12/site-packages/libtorrent/__init__.cpython-312-x86_64-linux-gnu.so'' (syntax-error)` complaint

**Comment by ngie-eign** [comment]
I'm running into the same problem with a project that doesn't distribute .py files when invoking pylint with `--unsafe-load-any-extension` and a custom `pylintrc`. I can provide this detail if needed.
FWIW, it looks like it may be due to some AST parsing/integration issues with asteroid:
https://github.com/pylint-dev/pylint/blob/d796007f3978469a7e6b3a5e883152c77d12e048/pylint/checkers/imports.py#L1061-L1071

**Comment by ngie-eign** [comment]
https://github.com/pylint-dev/astroid/issues/2733 looks very familiar.
